### ISSUE TITLE
option to set location for sourcemaps (for console logs)

### ISSUE THREAD

**Issue by farfromrefug** [NativeScript/nativescript-cli#5232]
Ok so to get sourcemaps fully working with Sentry, i have to modify quite a few things:
- change the location of sourcemaps (like with hidden sourcemaps)
- force `sourceMappingURL` to follow the format `app:///bundle.js.map` (so no mention to the relative path where source maps are located).

Consequently `sourceMapConverter` will fail [here](https://github.com/NativeScript/nativescript-cli/blob/4fa7e65071bb527a3b3784b1be79dd3d019667e4/lib/services/log-source-map-service.ts#L47) because this is the wrong location. I guess it tries to find the sourcemap in the same dir as the source itself. Or even what does it do with `app:///`

Do you think it would be possible to define somewhere (project wise) the location of the sourcemaps. May be using https://github.com/thlorenz/convert-source-map#frommapfilecommentcomment-mapfiledir ?

**Comment by rosen-vladimirov** [comment]
Hey @farfromrefug ,
In case the parsing of the source map fails (which probably happens in your scenario), the thing that will not be as usual are the logs CLI shows, i.e. they will not point to the real files, but instead will have stacktrace from bundle.js, vendor.js, etc.
Is this the issue you want to resolve with this feature request or something else is not working on your side?

**Comment by farfromrefug** [comment]
@rosen-vladimirov yes sorry for not being clear enough. Indeed what is failing for me when using sentry is the cli logging and sourcemap use

**Comment by bellalMohamed** [comment]
I uploaded the sourceMap file to sentry put unfortunately it didn't work, is there a specific --url-prefix I should use while uploading the .map files?

**Comment by farfromrefug** [comment]
Yes use app:///

**Comment by bellalMohamed** [comment]
@farfromrefug not working for me :(

**Comment by farfromrefug** [comment]
@bellalMohamed let s move that discussion to slack or discords. I am on both

**Comment by bellalMohamed** [comment]
> @bellalMohamed let s move that discussion to slack or discords. I am on both

Native script channel or private?

**Comment by farfromrefug** [comment]
whatever. Best to share it with others i guess

**Comment by vandammeb** [comment]
@farfromrefug @bellalMohamed Did you get this working in the end?

**Comment by bellalMohamed** [comment]
@vandammeb check this repo from @farfromrefug [https://github.com/farfromrefug/ns-forecastie](https://github.com/farfromrefug/ns-forecastie)

**Comment by Kolobok12309** [comment]
@bellalMohamed Can you help with envs in this repo)
Can't do work it without `SENTRY_PREFIX` and `SOURCEMAP_REL_DIR` envs

**Comment by farfromrefug** [comment]
@Kolobok12309 should ask that on Discord. You can find me there. I ll help
